### ISSUE TITLE
[bug]sqlite State is not properly controlled

### ISSUE THREAD

**Issue by hktalent** [dbeaver/dbeaver#16378]
1. Connect to the sqlite database in dbeaver, view the data, and show 1 entry
2. Then outside dbeaver, delete the database file, recreate the database file through other programs, and add two new data at the same time
3. Refresh the data in dbeaver, and always display only one data
4. Disconnect, reconnect, and refresh the data to display the two data correctly

**Comment by ShadelessFox** [comment]
Hello @hktalent,

The described behavior is driver-specific. In the case of SQLite, I believe it loads the entire database file into memory and caches it, so it doesn't need to access the filesystem. So that's why when recreating the database file does not affect the data. Perhaps modifying a row or two and then disconnecting from the data source will cause the database file to be flushed to the disk, likely overwriting or corrupting data.

In conclusion, this issue has nothing to do with DBeaver.
